### PR TITLE
Right-click support for WinUI

### PR DESCRIPTION
**Pull request by mvegaca**
# PR checklist
Review right click support for WinUI
 - Alow right-click in WinUI 3 C# Desktop
 - Alow right-click in WinUI 3 C++ Desktop
 - Alow right-click in WinUI 3 C++ UWP
 - Refactor ProjectInfoDataService improving the code.
 - Add WinUI verifications in infering data from ProjectInfoDataService
 - Allow projectMetadataService to save infered metadata in appxmanifest and XML metadata files

**Quick summary of changes**
Review right-click support for WinUI #4042

**Which issue does this PR relate to?**

**Applies to the following platforms:**

| UWP              | WPF              | WinUI           |
| :--------------- | :--------------- | :---------------|
| No | No | Yes |

### CODE REVIEW

File: code/src/UI/Services/ProjectConfigInfoService.cs [hunk 1]
```diff
@@ -443,9 +477,23 @@ private static bool ContainsNugetPackage(string packageId)
             return false;
         }
 
-        private static bool ContainsSDK(string sdkId)
+        private bool ContainsNugetPackageCpp(string packageId)
```
Comment (mrlacey) [inline code/src/UI/Services/ProjectConfigInfoService.cs:480]: Why this name? It doesn't appear to have anything specific to C++/CPP in it.

**Comment by mrlacey** [comment]
Is this PR meant to include the tests you mentioned [here](https://github.com/microsoft/WindowsTemplateStudio/pull/4077#issuecomment-789936726)?

File: code/src/UI/Services/ProjectConfigInfoService.cs (hunk 1 above)
Comment (mvegaca) [reply code/src/UI/Services/ProjectConfigInfoService.cs:480]: We look for added nuget in C# (csproj) and C++ (packages.config) in different files

Comment (sibille) [reply code/src/UI/Services/ProjectConfigInfoService.cs:480]: @mvegaca,  Can we call this ContainsNugetPackageInPackageConfig or do a method that's called ContainsNugetPackage and looks in both *.proj file and packages.config file? packages.config is not something specific to C++

**Comment by sibille** [comment]
> Is this PR meant to include the tests you mentioned here?

No, but I opened a separate issue for this.  

X-Ref: https://github.com/microsoft/WindowsTemplateStudio/issues/4116.

File: code/src/UI/Services/ProjectConfigInfoService.cs (hunk 1 above)
Comment (sibille) [reply code/src/UI/Services/ProjectConfigInfoService.cs]: Changed in https://github.com/microsoft/WindowsTemplateStudio/pull/4114/commits/8dfb2d94542631a98f8e69addbd63ed3711f4e1c after talking to @mvegaca

**Review by sibille — APPROVED** [review]

**Pull request merged**